src/main/java/com/snowflake/kafka/connector/internal/LoggerHandler.java

```diff
@@ -37,19 +27,41 @@ public static void setConnectGlobalInstanceId(UUID id) {
    *
    * @param id String attached to every log
    */
-  public static void setConnectGlobalInstanceId(String id) {
-    if (id != null && !id.isEmpty() && !id.equals(EMPTY_ID)) {
+  public static void setKcGlobalInstanceId(String id) {
+    if (id != null && !id.isEmpty()) {
       kcGlobalInstanceId = id;
       META_LOGGER.info(
           "Set Kafka Connect global instance id tag for logging: '{}'", kcGlobalInstanceId);
     } else {
-      META_LOGGER.info(
+      META_LOGGER.warn(
           "Given Kafka Connect global instance id was invalid (null or empty), continuing to log"
               + " without it");
-      kcGlobalInstanceId = EMPTY_ID;
+      kcGlobalInstanceId = "";
     }
   }
 
+  /**
+   * Returns the instance id as the hashcode of the kc start time
+   *
+   * @param startTime the start time
+   * @return the formatted instance id
+   */
+  public static String getFormattedKcGlobalInstanceId(long startTime) {
+    return Utils.formatString(KC_GLOBAL_INSTANCEID_FORMAT, Math.abs(("" + startTime).hashCode()));
```

Comment:
Just thinking out loud, cant we keep startTime instead of its hashcode?

Reply:
Also do we even need to keep startTime? The logger already prints the timestamp.

Reply:
I had the same thought - I don't think we need startTime since it has the timestamp. However it does easily provide us with an unique ID. I hashed it because the hash is shorter than the timestamp, to save space in the logs

Reply:
And that unique Id can also be used as a taskId right?